lib/condorcet_poll.rb

```diff
@@ -0,0 +1,60 @@
+require 'date'
+
+class CondorcetPoll
+  class InvalidCandidateError < StandardError
+  end
+
+  class DuplicatedVoterError < StandardError
+  end
+
+  class VoteToExpiredPollError < StandardError
+  end
+
+  attr_reader :title, :candidates, :expiresAt, :votes
+  def initialize(title, candidates, expiresAt = Date.today + 1)
+    @title = title
+    @candidates = candidates
+    @expiresAt = expiresAt
+    @votes = []
+  end
+
+  def add_vote(rankedVote)
+    unless rankedVote.orderedCandidates.sort == candidates.sort
+      raise InvalidCandidateError
+    end
+    if votes.map { |v| v.voter }.include?(rankedVote.voter)
+      raise DuplicatedVoterError
+    end
+    if @expiresAt < Date.today
+      raise VoteToExpiredPollError
+    end
+    @votes.push(rankedVote)
+  end
+
+  def get_winner
+    candidateWins = {}
+    @candidates.each { |candidate| candidateWins[candidate] = 0 }
+    @candidates.combination(2).each do |comb|
```

Comment:
[nit]

```suggestion
    @candidates.combination(2).each do |cand1, cand2|
```

と書けば、ブロックパラメータの配列をはじめから展開することができます。
cf. https://docs.ruby-lang.org/ja/3.0.0/doc/spec=2fcall.html#block_arg